python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1122,31 +1129,56 @@ def _ensure_persisted_execution_plan_snapshot(
 
         return execution_plan_snapshot_id
 
-    def _log_asset_materialization_planned_events(self, pipeline_run, execution_plan_snapshot):
+    def _log_asset_materialization_planned_events(
+        self, run: DagsterRun, execution_plan_snapshot: ExecutionPlanSnapshot
+    ):
         from dagster._core.events import (
             AssetMaterializationPlannedData,
             DagsterEvent,
             DagsterEventType,
         )
 
-        pipeline_name = pipeline_run.pipeline_name
+        pipeline_name = run.pipeline_name
 
         for step in execution_plan_snapshot.steps:
             if step.key in execution_plan_snapshot.step_keys_to_execute:
                 for output in step.outputs:
+                    assert output.properties
                     asset_key = output.properties.asset_key
                     if asset_key:
+                        partition_key = None
+
                         # Logs and stores asset_materialization_planned event
+                        partition_tag = run.tags.get(PARTITION_NAME_TAG)
+                        partition_range_start, partition_range_end = run.tags.get(
+                            ASSET_PARTITION_RANGE_START_TAG
+                        ), run.tags.get(ASSET_PARTITION_RANGE_END_TAG)
+
+                        check.invariant(
+                            not (partition_tag and partition_range_start),
+                            "Cannot have both a partition and a partition range",
+                        )
+
+                        if partition_tag:
+                            partition_key = partition_tag
+                        if partition_range_start:
+                            check.invariant(
+                                partition_range_end, "Partition range start set but not end"
+                            )
+                            # TODO: resolve which partitions are in the range, and emit an event for each
+
                         event = DagsterEvent(
                             event_type_value=DagsterEventType.ASSET_MATERIALIZATION_PLANNED.value,
                             pipeline_name=pipeline_name,
                             message=(
                                 f"{pipeline_name} intends to materialize asset"
                                 f" {asset_key.to_string()}"
                             ),
-                            event_specific_data=AssetMaterializationPlannedData(asset_key),
+                            event_specific_data=AssetMaterializationPlannedData(
+                                asset_key, partition_key=partition_key
```

Comment:
feel free to ignore, but could save a couple lines by doing `partition_key=partition_tag` here